R/data.table.R

```diff
@@ -686,10 +686,7 @@ replace_dot_alias = function(e) {
     if (is.null(jsub)) return(NULL)
 
     if (!with) {
-      if (jsub %iscall% ":=") {
-        # TODO(>=1.18.0): Simplify this error
-        stopf("with=FALSE together with := was deprecated in v1.9.4 released Oct 2014; this has been warning since v1.15.0. Please wrap the LHS of := with parentheses; e.g., DT[,(myVar):=sum(b),by=a] to assign to column name(s) held in variable myVar. See ?':=' for other examples.")
-      }
+      if (jsub %iscall% ":=") stopf("`:=` is only supported under with=TRUE.")
```

Comment:
Maybe keep the 
> See ?':=' for other examples."
although a bit superfluous?

Reply:
no, I think that's good. some users might also benefit from the nudge of how exactly to search for non-syntactic `:=` help, too.